website/source/docs/vault-integration/index.html.md

```diff
@@ -0,0 +1,154 @@
+
+```json
+{
+    "allowed_policies": "<comma-seperated list of policies>",
+    "explicit_max_ttl": 0,
+    "name": "nomad",
+    "orphan": false,
+    "period": 259200,
+    "renewable": true
+}
+```
+
+#### Parameters: 
+
+* `allowed_policies`: The `allowed_policies` is a comma separated list of
+  policies. This list should contain all policies that jobs running under Nomad
+  should have access to. Further, the list must contain one or more policies
+  that gives Nomad the following permissions:
+
+    ```
+    # Allow creating tokens under the role
+    path "auth/token/create/<role_name>" {
+        capabilities = ["create", "update"]
+    }
+
+    # Allow looking up the role
+    path "auth/token/roles/<role_name>" {
+        capabilities = ["read"]
+    }
+
+    # Allow looking up incoming tokens to validate they have permissions to
+    # access the tokens they are requesting
+    path "auth/token/lookup/*" {
+        capabilities = ["read"]
+    }
+
+    # Allow revoking tokens that should no longer exist
+    path "/auth/token/revoke-accessor/*" {
+        capabilities = ["update"]
+    }
+    ```
+
+* `explicit_max_ttl`: Must be set to `0` to allow periodic tokens.
+
+* `name`: Any name is acceptable. Replace `<role_name>` above with the chosen
+  name.
+
+* `orphan`: Must be set to `false`. This ensures that the token can be revoked
+  when the task is no longer needed or a node dies. This prohibits a leaked
+  token being used past the lifetime of a task.
+
+* `period`: Must be set to a positive value. The period specifies the length the
+  TTL is extended by each renewal in seconds. It is suggested to set this value
+  on the order of magniture of 3 days (259200 seconds) to avoid a large renewal
+  request rate to Vault.
+
+* `renewable`: Must be set to `true`. This is to allow Nomad to renew tokens for
+  tasks.
+
+See Vault's [Token Authentication Backend][auth] documentation for all possible
+fields and more complete documentation.
+
+#### Retrieving the Role based Token
+
+After the role is created, a token suitable for the Nomad servers may be
+retrieved by issuing the following Vault command:
+
+```
+$ vault token-create -role <role_name>
+Key             Value
+---             -----
+token           f02f01c2-c0d1-7cb7-6b88-8a14fada58c0
+token_accessor  8cb7fcb3-9a4f-6fbf-0efc-83092bb0cb1c
+token_duration  259200s
+token_renewable true
+token_policies  [<policies>]
+```
+
+The token can then be set in the Server configuration's [vault block][config] or
+as a command-line flag:
+
+```
+$ nomad agent -config /path/to/config -vault-token=f02f01c2-c0d1-7cb7-6b88-8a14fada58c0
+```
+
+## Agent Configuration
+
+To enable Vault integration, please see the [Nomad agent Vault
+integration][config] configuration.
+
+## Vault Definition Syntax
+
+To configure a job to retrieve Vault tokens, please see the [`vault` job
+specification documentation][vault-spec].
+
+## Assumptions
```

Comment:
Should there be a trouble shooting section? Will Nomad provide any debug output if the provided token is not valid or improperly setup?